CONTRIBUTING.md

```diff
@@ -23,12 +29,129 @@ tox run -e integration   # integration tests
+
+# Create the 'microk8s' group:
+newgrp microk8s
+
+# Enable the necessary Microk8s addons:
+microk8s enable hostpath-storage dns
 ```
 
-<!-- You may want to include any contribution/style guidelines in this document>
+#### Set up the Juju OLM
+```bash
+# Install the Juju CLI client, juju:
+sudo snap install juju --classic
+
+# Install a "juju" controller into your "microk8s" cloud:
+juju bootstrap microk8s airbyte-controller
+
+# Create a 'model' on this controller:
+juju add-model airbyte
+
+# Enable DEBUG logging:
+juju model-config logging-config="<root>=INFO;unit=DEBUG"
+
+# Check progress:
+juju status --relations --watch 2s
+juju debug-log
+```
+
+#### Deploy Charm
+```bash
+# Pack the charm:
+charmcraft pack [--destructive-mode]
+
+# Deploy the charm:
+juju deploy ./airbyte-k8s_ubuntu-22.04-amd64.charm --resource airbyte-api-server=airbyte/airbyte-api-server:0.60.0 --resource airbyte-bootloader=airbyte/bootloader:0.60.0 --resource airbyte-connector-builder-server=airbyte/connector-builder-server:0.60.0 --resource airbyte-cron=airbyte/cron:0.60.0 --resource airbyte-pod-sweeper=bitnami/kubectl:1.29.4 --resource airbyte-server=airbyte/server:0.60.0 --resource airbyte-worker=airbyte/worker:0.60.0
+
+# Deploy ui charm (Only if modifying UI charm, otherwise deploy using `juju deploy airbyte-ui-k8s --channel edge`)
+juju deploy ./airbyte-ui-k8s_ubuntu-22.04-amd64.charm --resource airbyte-webapp=airbyte/webapp:0.60.0
+```
+
+#### Relate Charms
+```bash
+# Relate operator to postgresql
+juju deploy postgresql-k8s --channel 14/edge --trust
+juju relate airbyte-k8s postgresql-k8s
+
+# Relate operator to minio
+juju deploy minio --channel edge
+juju relate airbyte-k8s minio
+
+# Deploy Temporal operators
+juju deploy temporal-k8s
+juju deploy temporal-admin-k8s
+juju relate temporal-k8s:db postgresql-k8s:database
+juju relate temporal-k8s:visibility postgresql-k8s:database
+juju relate temporal-k8s:admin temporal-admin-k8s:admin
+
+# Wait for units to settle and create default namespace
+juju run temporal-admin-k8s/0 tctl args="--ns default namespace register -rd 3"
+
+# Relate to Airbyte ui operator
+juju relate airbyte-k8s airbyte-ui-k8s
+
+# Generate private key
+openssl genrsa -out airbyte.key 2048
+
+# Generate a certificate signing request
+openssl req -new -key airbyte.key -out airbyte.csr -subj "/CN=airbyte-k8s"
+
+# Create self-signed certificate
+openssl x509 -req -days 365 -in airbyte.csr -signkey airbyte.key -out airbyte.crt -extfile <(printf "subjectAltName=DNS:airbyte-k8s")
+
+# Create a k8s secret
+kubectl create secret tls airbyte-tls --cert=airbyte.crt --key=airbyte.key
+
+# Deploy ingress controller
+microk8s enable ingress:default-ssl-certificate=airbyte/airbyte-tls
+
+# Deploy nginx operator
+juju deploy nginx-ingress-integrator --channel edge
+juju relate airbyte-ui-k8s nginx-ingress-integrator
+```
+
+#### Cleanup
+```bash
+# Clean-up before retrying
+juju destroy-model airbyte --destroy-storage
```

Comment:
You could add `juju destroy-application x` as a cleanup step. Someone may want to destroy the application but keep the model for redeployment (if they are making changes). Also I've found if you destroy the model with applications it is more likely to get buggy and get stuck in `destroying`